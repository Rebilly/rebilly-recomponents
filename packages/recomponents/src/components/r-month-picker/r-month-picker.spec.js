import {mount, shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RMonthPicker from './r-month-picker.vue';
import RIconButton from '../r-icon-button/r-icon-button.vue';

describe('r-month-picker.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RMonthPicker, {});
        expect(wrapper).toMatchSnapshot();
    });
    it('should render via SSR and match snapshot', async () => {
        const wrapper = await renderToString(RMonthPicker, {});
        expect(wrapper).toMatchSnapshot();
    });
    it('should select month and year if default values are set', async () => {
        const wrapper = await shallowMount(RMonthPicker, {
            propsData: {
                value: {monthIndex: 11, year: 2021},
            },
        });
        expect(wrapper.find('.r-month-picker-month-selected').exists()).toBeTruthy();
        expect(wrapper.find('.r-month-picker-month-selected').text()).toEqual('December');
        expect(wrapper.find('.r-month-picker-year').text()).toContain('2021');
    });
    it('should clear if clicked selected value and month picker is clearable', async () => {
        const wrapper = await shallowMount(RMonthPicker, {
            propsData: {
                value: {monthIndex: 11, year: 2021},
                clearable: true,
            },
        });

        expect(wrapper.find('.r-month-picker-month-selected').exists()).toBeTruthy();
        expect(wrapper.find('.r-month-picker-month-selected').text()).toEqual('December');
        try {
            await wrapper.find('.r-month-picker-month-selected').trigger('click');
        } finally {
            expect(wrapper.find('.r-month-picker-month-selected').exists()).toBeFalsy();
        }
    });
    it('should emit when year changes', async () => {
        const year = 2021;
        const wrapper = await mount(RMonthPicker, {
            propsData: {
                value: {monthIndex: 11, year},
            },
            components: {RIconButton},
        });
        await wrapper.find('.r-month-picker-year-left').trigger('click');
        expect(wrapper.emitted('input')).toBeTruthy();
        expect(wrapper.emitted('year-change')).toBeTruthy();
        expect(wrapper.emitted('year-change')).toEqual([[year - 1]]);
    });
});
