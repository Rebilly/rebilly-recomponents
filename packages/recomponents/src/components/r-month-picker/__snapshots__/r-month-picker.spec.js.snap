// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-month-picker.vue should render Wrapper and match snapshot 1`] = `
<div class="r-month-picker-container">
  <div class="r-month-picker-year">
    <r-icon-button-stub type="default" size="small" capitalizefirstletter="true">
      <r-icon-stub icon="caret-left"></r-icon-stub>
    </r-icon-button-stub>
    <div>2020</div>
    <r-icon-button-stub type="default" size="small" capitalizefirstletter="true">
      <r-icon-stub icon="caret-right"></r-icon-stub>
    </r-icon-button-stub>
  </div>
  <div class="r-month-picker">
    <div class="r-month-picker-month">
      January
    </div>
    <div class="r-month-picker-month">
      February
    </div>
    <div class="r-month-picker-month">
      March
    </div>
    <div class="r-month-picker-month">
      April
    </div>
    <div class="r-month-picker-month">
      May
    </div>
    <div class="r-month-picker-month r-month-picker-month-selected">
      June
    </div>
    <div class="r-month-picker-month">
      July
    </div>
    <div class="r-month-picker-month">
      August
    </div>
    <div class="r-month-picker-month">
      September
    </div>
    <div class="r-month-picker-month">
      October
    </div>
    <div class="r-month-picker-month">
      November
    </div>
    <div class="r-month-picker-month">
      December
    </div>
  </div>
</div>
`;

exports[`r-month-picker.vue should render via SSR and match snapshot 1`] = `
<div class="r-month-picker-container">
  <div class="r-month-picker-year"><button class="r-button r-button-type-default r-button-size-small r-has-icon">
      <div role="button" class="r-capitalize-first-letter button"><svg class="r-icon r-icon-20">
          <use xlink:href="#icon-caret-left" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
        </svg></div>
    </button>
    <div>2020</div> <button class="r-button r-button-type-default r-button-size-small r-has-icon">
      <div role="button" class="r-capitalize-first-letter button"><svg class="r-icon r-icon-20">
          <use xlink:href="#icon-caret-right" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
        </svg></div>
    </button>
  </div>
  <div class="r-month-picker">
    <div class="r-month-picker-month">
      January
    </div>
    <div class="r-month-picker-month">
      February
    </div>
    <div class="r-month-picker-month">
      March
    </div>
    <div class="r-month-picker-month">
      April
    </div>
    <div class="r-month-picker-month">
      May
    </div>
    <div class="r-month-picker-month r-month-picker-month-selected">
      June
    </div>
    <div class="r-month-picker-month">
      July
    </div>
    <div class="r-month-picker-month">
      August
    </div>
    <div class="r-month-picker-month">
      September
    </div>
    <div class="r-month-picker-month">
      October
    </div>
    <div class="r-month-picker-month">
      November
    </div>
    <div class="r-month-picker-month">
      December
    </div>
  </div>
</div>
`;
